### PR TITLE
Update to latest WalletConnect v1.x SDK #3489

### LINKED ISSUE #3489: Update to latest WalletConnect v1.x SDK
Check diffs against `1.3.1` which we are currently using https://github.com/WalletConnect/WalletConnectSwift/releases. Some changes mentioned might break compatibility?

> Add new Server and Client delegate methods

> v1.4.0 update CryptoSwift version

If it is too much breakage, we should consider not upgrading at all.

### PR DESCRIPTION
**Pull request by vladyslav-iosdev**
Closes #3489

### CODE REVIEW

File: AlphaWallet/WalletConnect/WalletConnectServer.swift [hunk 1]
```diff
@@ -382,6 +344,53 @@ extension WalletConnectServer: ServerDelegate {
...
+
+    private func removeSession(for url: WalletConnectURL) {
+        guard var sessions = _sessions.value else { return }
+
+        if let index = sessions.firstIndex(where: { $0.url.absoluteString == url.absoluteString }) {
+            set(server: nil, for: sessions[index].url)
+            sessions.remove(at: index)
+        }
+
+        UserDefaults.standard.walletConnectSessions = sessions
```
Comment (hboon) [inline AlphaWallet/WalletConnect/WalletConnectServer.swift:375]: Better to store a Config instance and read its `defaults` to have the correct `UserDefaults` instance?

**Review by hboon — APPROVED** [review]
Seems ok. Do you foresee any problem with upgrading `CryptoSwift`? It was quite painful the last time it was upgraded, right?

**Comment by hboon** [comment]
@vladyslav-iosdev BTW, can you verify that `Podfile.lock` was committed correct? I get a different checksum after I do `pod install`.

**Comment by vladyslav-iosdev** [comment]
> Do you foresee any problem with upgrading CryptoSwift?

 it was need to update because of `Wallet Connnect` requires  `CryptoSwift` greater then `~> 1.4`

**Review by hboon — APPROVED** [review]

**Pull request merged**